### PR TITLE
overlord/ifacestate: tweak logic for generating unique slot names

### PR DESCRIPTION
**Pull request by stolowski**
Based on #6465, the actual change is limited to ensureUniqueName and tests (TestHotplugSlotName, TestEnsureUniqueName).

Modified the logic of ensureUniqueName not to treat the last part of the input name as numeric suffix (and increment it to ensure uniqueness) - instead the input is treated as a base name that's not touched, to preserve names such as "ieee1394"(with the old code this could become "ieee1395" if there was a "ieee1394" slot already); a "-1", "-2" etc suffix is appended to the base for uniqueness.

The old code was unnecessarily trying to be too smart with this due to the wrong assumption that we would need to generate names based on already generated names, which is not the case (we always have raw device attributes and a base/suggested name coming from interface code).

### CODE REVIEW

File: overlord/ifacestate/hotplug_test.go [hunk 1]
```diff
@@ -238,15 +240,15 @@ func (s *hotplugSuite) TestEnsureUniqueName(c *C) {
 
 	names := []struct{ proposedName, resultingName string }{
 		{"foo", "foo"},
-		{"slot", "slot2"},
-		{"slot1", "slot2"},
-		{"slot1234", "slot1235"},
-		{"slot-1", "slot-3"},
-		{"slot3-5", "slot3-7"},
+		{"slot", "slot-3"},
```
Comment (pedronis) [inline overlord/ifacestate/hotplug_test.go]: it feels like this tests could do with some more tweaking given that realistic input is more like "ifacexyz" or "devname" than "slot123"

**Comment by pedronis** [comment]
Btw while this is a better starting point, I noticed looking at the other branch that we let interfaces suggest a name too via the Spec, so this might need still to evolve a bit, get some tweaks once we start writing actual hotplug support in interfaces with their naming code.

File: overlord/ifacestate/hotplug_test.go (hunk 1 above)
Comment (stolowski) [reply overlord/ifacestate/hotplug_test.go]: Done.

File: overlord/ifacestate/hotplug.go [hunk 2]
```diff
@@ -98,26 +97,22 @@ func (m *InterfaceManager) hotplugEnumerationDone() {
...
 // ensureUniqueName modifies proposedName so that it's unique according to isUnique predicate.
-// Uniqueness is achieved by appending a numeric suffix, or increasing existing suffix.
+// Uniqueness is achieved by appending a numeric suffix.
 func ensureUniqueName(proposedName string, isUnique func(string) bool) string {
 	// if the name is unique right away, do nothing
 	if isUnique(proposedName) {
 		return proposedName
 	}
 
-	suffixNumValue := 0
```
Comment (mvo5) [inline overlord/ifacestate/hotplug.go]: I like how simple this became :)

**Pull request merged**